src/index.ts

```diff
@@ -103,60 +59,34 @@ export class ThemeResolverPlugin {
         );
     }
 
-    public resolveComponentPath(reqPath: string): Promise<string> {
-        if (!this.cache[reqPath]) {
-            if (this.chosenResolver.directories) {
-                this.cache[reqPath] = Promise.filter(
-                    this.chosenResolver.directories.map((dir: string) => path.resolve(path.resolve(dir), reqPath)),
-                    (item: string) => existsAsync(item).then((exists: boolean) => exists).catch(() => false),
-                ).any();
-            } else {
-                this.cache[reqPath] = Promise.reject(new Error("No Fallback directories!"));
-            }
+    public resolveComponentPath(reqPath: string, directories: string[]): string | undefined {
+
+        if (this.cache[reqPath] !== undefined) {
+            return this.cache[reqPath]
         }
-        return this.cache[reqPath];
+
+        const dirs = directories.map((dir: string) => path.resolve(path.resolve(dir), reqPath))
+
+        const resolvedPath = dirs.find((path: string) => fs.existsSync(path))
+
+        if (resolvedPath) {
+            this.cache[reqPath] = resolvedPath
+        }
+
+        return this.cache[reqPath]
     }
 
-    public resolveComponentModule(reqPath: string): Promise<string> {
-        if (this.chosenResolver.module) {
-            if (this.chosenResolver.singlePackage) {
-                const tempReqPath = "node_modules/"
-                                    + this.chosenResolver.module
-                                    + this.chosenResolver.modulePath
-                                    + "/"
-                                    + reqPath;
-
-                this.cache[reqPath] = new Promise(
-                    (resolve: any, reject: any) => {
-                        try {
-                            const res =  path.resolve(process.cwd(), tempReqPath);
-                            if (res) {
-                                resolve(res);
-                            }
-                        } catch (e) {
-                        reject(new Error("Module is not resolvable"));
-                        }
-                    },
-                );
-            } else {
-                const dep = this.chosenResolver.module + "." + reqPath.split(".")[0];
-                this.cache[reqPath] = new Promise(
-                    (resolve: any, reject: any) => {
-                        try {
-                            const res = moduleresolver.sync(dep, {basedir: process.cwd()});
-                            if (res) {
-                                resolve(res);
-                            }
-                        } catch (e) {
-                        reject(new Error("Module is not resolvable"));
-                        }
-                    },
-                );
+    private getResolver(request: any): IThemeResolverPluginOptions | void
```

Comment:
Same here. If we figure out the type of `request`, it should apply here as well.